.github/workflows/pr-merged.yml

```diff
@@ -0,0 +1,177 @@
+      # This ensures we get the latest scripts, even if the PR's base was outdated.
+      - name: Checkout base branch
+        id: checkout
+        uses: actions/checkout@v4
+        # TODO: sparse checkout
+
+      - name: Install Nix
+        uses: cachix/install-nix-action@v31
+        with:
+          extra_nix_config: |
+            accept-flake-config = true
+
+      # NOTE: This is here for debugging and gathering data
+      # TODO: Remove
+      - name: Print metadata
+        env:
+          checked_out: ${{ steps.checkout.outputs.commit }}
+          workflow_base: ${{ github.base_ref }}
+          workflow_head: ${{ github.head_ref }}
+          parent: ${{ steps.base.outputs.sha }}
+          base: ${{ github.event.pull_request.base.sha }}
+          head: ${{ github.event.pull_request.head.sha }}
+          merge: ${{ github.event.pull_request.merge_commit_sha }}
+          pull_request: ${{ toJSON(github.event.pull_request) }}
+        run: |
+          {
+            echo '## Commits'
+            echo
+            echo "Checked out: $checked_out"
+            echo "Workflow base: $workflow_base"
+            echo "Workflow head: $workflow_head"
+            echo "PR parent: $parent"
+            echo "Base: $base"
+            echo "Head: $head"
+            echo "Merge: $merge"
+            echo
+            echo '<details><summary><code>pull_request</code> details</summary>'
+            echo
+            echo '```json'
+            echo "$pull_request" | jq .
+            echo '```'
+            echo
+            echo '</details>'
+            echo
+          } >> "$GITHUB_STEP_SUMMARY"
+
+      - name: Compare parent -> head
+        id: diff1
+        env:
+          base: ${{ steps.base.outputs.sha }}
+          head: ${{ github.event.pull_request.head.sha }}
+        run: |
+          out=$(mktemp)
+
+          # FIXME: first run will print to stdout: "pre-commit installed at .git/hooks/pre-commit"
+          nix develop --command echo first run
+
+          # FIXME: evaluating & building the devshell is expensive, we should
+          # provide an alternative way for CI to run diff-plugins
+          nix develop --command diff-plugins --compact "$base" "$head" > "$out"
+
+          {
+            echo -n 'json='
+            jq -c . < "$out"
+          } >> "$GITHUB_OUTPUT"
+          jq . < "$out"
+
+      - name: Compare base -> merge
+        id: diff2
+        env:
+          base: ${{ github.event.pull_request.base.sha }}
+          head: ${{ github.event.pull_request.merge_commit_sha }}
+        run: |
+          out=$(mktemp)
+
+          # FIXME: first run will print to stdout: "pre-commit installed at .git/hooks/pre-commit"
+          nix develop --command echo first run
+
+          # FIXME: evaluating & building the devshell is expensive, we should
+          # provide an alternative way for CI to run diff-plugins
+          nix develop --command diff-plugins --compact "$base" "$head" > "$out"
```

Comment:
If we want to avoid building the entire dev shell, we can just make the `diff-plugins` use a `nix-shell` shebang and call it directly.

Reply:
created https://github.com/nix-community/nixvim/pull/3516 to show you what i mean